src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5582,37 +5548,34 @@ int PlatformEvent::park(jlong Millis) {
   if (v != 0) return OS_OK;
 
   // Do this the hard way by blocking ...
-  // TODO: consider a brief spin here, gated on the success of recent
-  // spin attempts by this thread.
-  //
   // We decompose long timeouts into series of shorter timed waits.
   // Evidently large timo values passed in WaitForSingleObject() are problematic on some
   // versions of Windows.  See EventWait() for details.  This may be superstition.  Or not.
-  // We trust the WAIT_TIMEOUT indication and don't track the elapsed wait time
-  // with os::javaTimeNanos().  Furthermore, we assume that spurious returns from
-  // ::WaitForSingleObject() caused by latent ::setEvent() operations will tend
-  // to happen early in the wait interval.  Specifically, after a spurious wakeup (rv ==
-  // WAIT_OBJECT_0 but _Event is still < 0) we don't bother to recompute Millis to compensate
-  // for the already waited time.  This policy does not admit any new outcomes.
-  // In the future, however, we might want to track the accumulated wait time and
-  // adjust Millis accordingly if we encounter a spurious wakeup.
 
   const int MAXTIMEOUT = 0x10000000;
-  DWORD rv = WAIT_TIMEOUT;
-  while (_Event < 0 && Millis > 0) {
-    DWORD prd = Millis;     // set prd = MAX (Millis, MAXTIMEOUT)
+  jlong waitStart = os::javaTimeNanos();
+  // preserve initial millis for remaining time calculation on spurious wakeup
+  jlong initialMillis = Millis;
+  BOOL rv;
+  while ((v = _Event) < 0 && Millis > 0) {
+    DWORD prd = Millis;     // set prd = MIN (Millis, MAXTIMEOUT)
     if (Millis > MAXTIMEOUT) {
       prd = MAXTIMEOUT;
     }
     HighResolutionInterval *phri = nullptr;
```

Comment:
@djelinski Is this even used?

Reply:
yeah, it's the C++ construction where the constructor and the destructor have side effects. It increases the system timer resolution, unless `ForceTimeHighResolution` is set. `ForceTimeHighResolution`, contrary to its name and help text, forces [low timer resolution](https://bugs.openjdk.org/browse/JDK-6435126). Sigh.

Reply:
Oh. That's.... interesting. 😞

Reply:
Of course pre-existing, but the typical pattern we use with RAII objects that may or may not do something is to give it a constructor argument, e.g. `bool activate`, that controls if it is activated. Allocating an RAII object with new/delete seems odd.

Reply:
Please just let this sleeping dog lie. ;-)